### ISSUE TITLE
Example code for process_packets.py needs better handling of packet object attributes

### ISSUE THREAD

**Issue by pauldalewilliams** [XSEDE/amieclient#14]
The example code in [process_packets.py](https://github.com/XSEDE/amieclient/blob/main/examples/process_packets.py) needs better handling of variable assignment from the packet object attributes.  As it is written Python will raise an exception if an attribute doesn't exist, which is possible when a packet did not contain an allowed data key.

I think the best option is to use [getattr](https://docs.python.org/3.9/library/functions.html#getattr) with None as the default.  Alternatively, the packet could get converted to a dictionary first with the as_dict method and the values retrieved with the get method on the resulting dictionary.

Another option would be to change the behavior of the Packet class's [__init__ function here](https://github.com/XSEDE/amieclient/blob/main/amieclient/packet/base.py#L167) so that all keys are at least present as an attribute on the object.  I'm not sure that would be a desired change though.

Just a suggestion to help make the example code something that people can run with rather than hitting errors right out of the gate.

**Comment by sabo** [comment]
Hi @pauldalewilliams, thanks for taking a look. We actually handle attributes that represent fields of the packet sort of like your last suggestion -- if you look [here](https://github.com/XSEDE/amieclient/blob/9b1932b39d2ed788875ec573152d8901e8862d34/amieclient/packet/base.py#L56), we use a metaclass to build each individual packet class. We can't do it in `__init__()` since (at least in the versions of python we have to support) there's actually no good way that I found to dynamically create property attributes at the class level when instantiating an object in that class. 

The metaclass's `__new__()` function run when the individual packet classes are defined, takes a look at the fields specified in the declaration of each packet class (for instance, [RequestProjectCreate here](https://github.com/XSEDE/amieclient/blob/9b1932b39d2ed788875ec573152d8901e8862d34/amieclient/packet/project.py#L141)) and defines them as properties with special setters and getters. Each field's [getter methods](https://github.com/XSEDE/amieclient/blob/9b1932b39d2ed788875ec573152d8901e8862d34/amieclient/packet/base.py#L37-L40)  do in fact return None if they haven't been set otherwise, since they're using `get()` on a dict behind the scenes. 

You can see this in action just by creating an empty packet and  checking an arbitrary attribute:

```python
In [1]: import amieclient                                                                                                                                                                           

In [2]: p = amieclient.packet.RequestAccountCreate()                                                                                                                                                
                                                                                                                                                           
In [3]: p.AcademicDegree is None                                                                                                                                                                    
Out[3]: True
```

**Comment by pauldalewilliams** [comment]
Ah, so in digging to recreate the error I first encountered a couple months ago when trying the example code I think I found what's wrong.  The process_packets.py example attempts to assign packet.PiRemoteSiteLogin to a variable on [this line](https://github.com/XSEDE/amieclient/blob/main/examples/process_packets.py#L41), but that's not in the [data_keys for RPC packets](https://github.com/XSEDE/amieclient/blob/main/amieclient/packet/project.py#L141).  The same is true for PiNsfStatusCode.

Looking at the raw packets it appears that those PiNsfStatusCode should just be NsfStatusCode.  Looking at the old AMIE docs I'm not sure PiRemoteSiteLogin is expected in a RPC packet anyway, so maybe that line just needs to be removed from the example.

Apologies for the red herring there - it does look like those two lines should be adjusted in the example code though.
